lib/randomstring.js

```diff
@@ -13,31 +13,55 @@ function safeRandomBytes(length) {
   }
 }
 
-exports.generate = function(options) {
-  
+function processString(buf, initialString, chars, reqLen, maxByte) {
+  var string = initialString;
+  for (var i = 0; i < buf.length && string.length < reqLen; i++) {
+    var randomByte = buf.readUInt8(i);
+    if (randomByte < maxByte) {
+      string += chars.charAt(randomByte % chars.length);
+    }
+  }
+  return string;
+}
+
+function getAsyncString(string, chars, length, maxByte, cb) {
+  crypto.randomBytes(length, function(err, buf) {
+    if (err) {
+      getAsyncString(string, chars, length, maxByte, cb);
```

Comment:
Unlike the sync version which documents clearly what the error will be--I don't see any explanation of what this error could be. I'm not sure it's safe to retry.